R/count_teleconnections.R

```diff
@@ -55,6 +55,101 @@ count_watershed_teleconnections <- function(data_dir,
   import_raster(paste0(data_dir, crop_file_path)) ->
     cropcover_USA
 
+      # START GCAM CROP CLASS BINDING
+
+      # Load in the GCAM classes CSV.
+      gcam_csv <- get_crop_mapping()
+
+      # Delete the rows that have crop types not present in CDL classes.
+      gcam_csv <- gcam_csv[-c(70, 100, 131, 45, 104, 182, 181),]
+
+      # Extract only the GCAM Class column
+      as.data.frame(gcam_csv[,c(3)]) %>%
+        mutate(GCAM_ID = group_indices_(gcam_csv, .dots=c("GCAM_commodity"))) ->
+        gcam_classes
+
+      # Shrink the Column so that there is only one of each Class
+      gcam_class <- distinct(gcam_classes)
+
+      # Obtain the attribute table for the raster.
+      levels(cropcover_USA)[[1]] %>%
+        filter(Class_Names != "") %>%
+        dplyr::select(ID, CDL_Class = Class_Names) %>%
+        mutate(CDL_Class = as.character(CDL_Class))->
+            cdl_classes
+
+      # This HUGE chunk assigns the GCAM IDs to matching CDL Class. Most likely a more efficient way of doing this.
```

Comment:
I am on an old laptop and cannot test this right now but try using the `recode` function from the `car` package (not to be confused with the `dplyr` one) in the following simple (fake) example:

```r
# fake matrix
m <- matrix(c(3:14), nrow = 4, byrow = TRUE)

# recoding all at once with assignment and ranges
car::recode(m, "3=0; c(5, 6, 7)=100; 8:14=107")
```

You will still need to build the lookup string but it should be more concise.  Consider constructing it from an input file so the values in the file can be changed but the process is consistent.  I will try to take a look at this in detail this weekend - I'll be in and out of the office tomorrow.  There may still be a more efficient way.